arangod/IResearch/IResearchLink.cpp

```diff
@@ -367,6 +368,24 @@ void IResearchLink::afterTruncate(TRI_voc_tick_t tick) {
 
   TRI_ASSERT(_dataStore);  // must be valid if _asyncSelf->get() is valid
 
+  if (trx != nullptr) {
+    auto* key = this;
+
+    auto& state = *(trx->state());
+
+    // TODO FIXME find a better way to look up a ViewState
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    auto* ctx = dynamic_cast<LinkTrxState*>(state.cookie(key));
+#else
+    auto *ctx = static_cast<LinkTrxState*>(state.cookie(key));
```

Comment:
For consistency, it should be:
```suggestion
    auto* ctx = static_cast<LinkTrxState*>(state.cookie(key));
```